frontend/src/pages/notebookController/screens/server/StartServerModal.tsx

```diff
@@ -53,32 +63,46 @@ const StartServerModal: React.FC<StartServerModalProps> = ({ open, spawnInProgre
   const spawnFailed =
     spawnStatus?.status === AlertVariant.danger || spawnStatus?.status === AlertVariant.warning;
 
+  const navigateToNotebook = React.useCallback(
+    (useCurrentTab: boolean): void => {
+      if (notebookLink) {
+        if (useCurrentTab) {
+          window.location.href = notebookLink;
+        } else {
+          window.open(notebookLink, '_blank');
+          history.push('/notebookController');
```

Comment:
Do we need this? 🤔 This is because they can just close the modal?

Reply:
I can remove this force redirect, but I may need to change the logic to set `showClose` of the Modal. Currently, we only show the close icon when the spawning fails, that's how they did in JH, so there is no way for the users to close the modal if the spawning succeeds. We can definitely change that to show the close icon on both success (without setting spawn in the current tab) and failed status. WDYT?

Reply:
No definitely don't show additional buttons unless we need to, it doesn't need to be a nosy modal of buttons. But why did you add it in the first place? Am I missing the use case here -- don't we just stop the redirect while the page is going or was that logic removed somewhere? 🤔

Reply:
Let's think of the scenarios. First, on the spawner page, the user doesn't check `Start server in current tab`, then he starts a notebook, when the spawning success, we will show the modal, and there are 2 buttons for the user to choose whether to open a new tab for the notebook or launch the notebook in the current tab. What if he chooses to launch the notebook in a new tab? Then there will be a new tab for the notebook opens, but the old tab is still there, when he switches back to the old tab (dashboard one), he can still see this starting server modal and cannot even find a button to close it. I am doing this because when the user makes the choice to open the notebook whether in a new tab or a current tab, I should help him to navigate to the controller page for the dashboard old tab, but not stay watching the un-closable modal.

Reply:
Wow, yes -- of course. These were comments made by me without having fired up your PR. Absolutely correct, thanks for the clarification.